src/reader/mod.rs

```diff
@@ -185,11 +185,10 @@ impl JournalReader {
 		{
 			unsafe {
 				let b = ::std::slice::from_raw_parts(data, sz as usize);
-				let field = String::from_utf8_lossy(b);
-				let mut name_value = field.splitn(2, '=');
-				let name = name_value.next().unwrap();
-				let value = name_value.next().unwrap();
-				fields.insert(From::from(name), From::from(value));
+				let mut name_value = b.splitn(2, |x| *x == b'=');
+				let name = String::from_utf8_lossy(name_value.next().unwrap()).into_owned();
+				let value = name_value.next().unwrap().to_vec();
+				fields.insert(name, value);
```

Comment:
I think only the `from_raw_parts` call is unsafe. It would be nice to move the rest out of the unsafe block to make it clear what the unsafe bit is.